<launch>
  <!-- Define arguments for the initial pose -->
  <arg name="spawn_x" default="-5.00796"/>
  <arg name="spawn_y" default="2.33341"/>
  <arg name="spawn_z" default="0.169994"/>
  <arg name="spawn_roll" default="0.000035"/>
  <arg name="spawn_pitch" default="0"/>
  <arg name="spawn_yaw" default="4.71"/>

  <!-- Publish Initial Pose -->
  <node pkg="rostopic" type="rostopic" name="set_initial_pose"
        args="pub -1 /initialpose geometry_msgs/PoseWithCovarianceStamped '{header: {frame_id: map}, pose: {pose: {position: {x: $(arg spawn_x), y: $(arg spawn_y), z: $(arg spawn_z)}, orientation: {z: 0.0, w: 0.0}}, covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]}}'" />
  
  <!-- Launch RViz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find p3at_description)/configs/navigation.rviz"/>

  
</launch>

